week3/fr0gydev/todo/app.js

```diff
@@ -0,0 +1,121 @@
+const todoListEl = document.getElementById("todoList");
+const todoInputEl = document.getElementById("todoInput");
+
+const API_URL = "http://localhost:8080/todos";
+
+fetch(API_URL)
+  .then((response) => {
+    if (!response.ok) throw new Error('Network response was not ok');
+    return response.json();
+  })
+  .then((data) => renderTodo(data))
+  .catch((error) => {
+    console.error("Error fetching todos:", error);
+    todoListEl.innerHTML = "<li>Failed to load todos. Please check your server connection.</li>";
+  });
+
+const updateTodo = (todoId, originalTitle) => {
+  const newTitle = prompt("Enter new title for the todo:", originalTitle);
+  
+  if (newTitle !== null && newTitle !== originalTitle) {
```

Comment:
```javascript
newTitle != null
```
이렇게 체크하면 null, undefined도 같이 체크할 수 있어서 이 편이 더 좋습니다

Reply:
```suggestion
  const isTitleChanged = newTitle !== originalTitle
  const isNewTitleNotNil = newTitle != null
  
  if (isNewTitleNotNil && isTitleChanged) {
```
요렇게 둘 다 변수로 분리하는 것이 더 좋습니다